wiring/src/spark_wiring_ble.cpp

```diff
@@ -2378,6 +2378,12 @@ class BleScanDelegator {
      */
     static void onScanResultCallback(const hal_ble_scan_result_evt_t* event, void* context) {
         BleScanDelegator* delegator = static_cast<BleScanDelegator*>(context);
+
+        if (!delegator->filter_.allowDuplicates() && delegator->isCachedDevice(event->peer_addr)) {
+            return;
+        }
+        delegator->cachedDevices_.append(event->peer_addr);
```

Comment:
This should not be populated in case of `allowDuplicates()` to avoid uncontrollable growth of the vector, as otherwise it will continue to fill-up with duplicates.

Same can potentially happen normally depending on scan duration, although highly unlikely, but should potentially be taken care of somehow as well.